doctr/datasets/imgur5k.py

```diff
@@ -0,0 +1,100 @@
+
+__all__ = ["IMGUR5K"]
+
+
+class IMGUR5K(AbstractDataset):
+    """IMGUR5K dataset from `"TextStyleBrush: Transfer of Text Aesthetics from a Single Example"
+    <https://arxiv.org/abs/2106.08385>`_ |
+    `"repository" <https://github.com/facebookresearch/IMGUR5K-Handwriting-Dataset>`_.
+
+    Example::
+        >>> # NOTE: You need to download/generate the dataset from the repository.
+        >>> from doctr.datasets import IMGUR5K
+        >>> train_set = IMGUR5K(train=True, img_folder="/path/to/IMGUR5K-Handwriting-Dataset/images",
+        >>>                     label_path="/path/to/IMGUR5K-Handwriting-Dataset/dataset_info/imgur5k_annotations.json")
+        >>> img, target = train_set[0]
+        >>> test_set = IMGUR5K(train=False, img_folder="/path/to/IMGUR5K-Handwriting-Dataset/images",
+        >>>                    label_path="/path/to/IMGUR5K-Handwriting-Dataset/dataset_info/imgur5k_annotations.json")
+        >>> img, target = test_set[0]
+    Args:
+        img_folder: folder with all the images of the dataset
+        label_path: path to the annotations file of the dataset
+        use_polygons: whether polygons should be considered as rotated bounding box (instead of straight ones)
+    """
+
+    def __init__(
+        self,
+        img_folder: str,
+        label_path: str,
+        train: bool = True,
+        use_polygons: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(img_folder, pre_transforms=convert_target_to_relative, **kwargs)
+
+        # File existence check
+        if not os.path.exists(label_path) or not os.path.exists(img_folder):
+            raise FileNotFoundError(
+                f"unable to locate {label_path if not os.path.exists(label_path) else img_folder}")
+
+        self.data: List[Tuple[Path, Dict[str, Any]]] = []
+        self.train = train
+        np_dtype = np.float32
+
+        img_names = os.listdir(img_folder)
+        train_samples = int(len(img_names) * 0.9)
+        set_slice = slice(train_samples) if self.train else slice(train_samples, None)
+
+        with open(label_path) as f:
+            annotation_file = json.load(f)
+
+        for img_name in img_names[set_slice]:
+            img_path = Path(img_folder, img_name)
+            img_id = img_name.split(".")[0]
+
+            # File existence check
+            if not os.path.exists(os.path.join(self.root, img_name)):
+                raise FileNotFoundError(f"unable to locate {os.path.join(self.root, img_name)}")
+
+            # some files have no annotations
+            if img_id not in annotation_file['index_to_ann_map'].keys():
+                continue
+            ann_ids = annotation_file['index_to_ann_map'][img_id]
+            annotations = [annotation_file['ann_id'][a_id] for a_id in ann_ids]
+
+            labels: List[str] = []
+            box_targets: List[np.ndarray] = []
+            for ann in annotations:
+                if ann['word'] != '.':
```

Comment:
Does that mean that the text "final." will have two boxes?

Reply:
@fg-mindee 
no there are some wrong/missing annotations in the set which have only a dot in dict key 'word'
https://github.com/facebookresearch/IMGUR5K-Handwriting-Dataset/blob/main/README.md
Bounding boxes with '.' mean the annotations were not done for various reasons

Example from annotations:
```
"qNgR61W_99": {
            "word": ".",
            "bounding_box": "[479.0, 496.0, 45.0, 35.0, -3.0]"
        },
        "qNgR61W_100": {
            "word": ".",
            "bounding_box": "[529.0, 498.0, 52.0, 37.0, 0.0]"
        },
        "qNgR61W_101": {
            "word": ".",
            "bounding_box": "[583.0, 494.0, 42.0, 37.0, -2.0]"
        },
```

Reply:
yep :)